pkg/build/tempdirs.go

```diff
@@ -60,10 +62,7 @@ func assertSourceHash(p crt.Product) {
 	}
 	// It's the maintainers' jobs to make sure we don't hit this panic.
 	// It's here to avoid writing undiscoverable files to the cache.
-	if (p == crt.Product{}) {
-		log.Panicf("SourceHash is empty; Product is empty.")
-	}
-	log.Panicf("SourceHash is empty; Product is nonempty: % #v", p)
```

Comment:
This assertion required that `crt.Product` did not contain a slice (i.e. for it to be comparable). It's not really a necessary check anymore, since issues with the SourceHash would be flagged up by many different tests anyway.